Move time_utc functions from AP_HAL to AP_Common

rmackay9 opened the pull request:
Moves the newly added get_ms_until_time_utc functions from AP_HAL to AP_Common/AP_Time.

This doesn't yet resolve the compile issue for Linux boards that has something to do with a missing library (-lrt ?)

lucasdemarchi commented:
First thing that comes to mind is that it would be better if they were static, so that we can use just like AP_HAL::panic().

lucasdemarchi commented:
There are several other functions related to time that could be moved to the same place. AP_HAL::millis() for example.

lucasdemarchi commented:
Ok... what if instead we move these functions from Util to AP_HAL and then don't touch millis() and the other ones?

khancyr commented on libraries/AP_Common/AP_Time.cpp:
```diff
@@ -0,0 +1,84 @@
+ */
+
+#include "AP_Time.h"
+#include <time.h>
```
Should'nt this include be in AP_Time.h ?

khancyr commented on libraries/AP_Common/AP_Time.h:
```diff
@@ -0,0 +1,18 @@
+ */
+
+#ifndef AP_TIME_H
+#define AP_TIME_H
```
Prefere #pragma once instead of include gard to match with other files

magicrub replied:
Agreed, should be pragma once.
On May 19, 2016 6:24 AM, "khancyr" <notifications@github.com> wrote:

> In libraries/AP_Common/AP_Time.h
> <https://github.com/ArduPilot/ardupilot/pull/4163#discussion_r63876702>:
>
> > @@ -0,0 +1,18 @@
> > +/*
> > + * AP_Time.h
> > + */
> > +
> > +#ifndef AP_TIME_H
> > +#define AP_TIME_H
>
> Prefere #pragma once instead of include gard to match with other files
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ArduPilot/ardupilot/pull/4163/files/fcd2a99db974b1fe70dd516b86d6e3af093a1198#r63876702>
>

rmackay9 commented:
Ok, fixed a couple of the issues pointed out, rebased and we shall see if it gets through travis or not.
@lucasdemarchi , @tridge  maybe you two could agree on where you want these functions to be?

OXINARF commented on libraries/AP_Common/AP_Time.h:
```diff
@@ -2,10 +2,10 @@
+#pragma once
 
 #include <AP_Common/AP_Common.h>
+#include <time.h>
```
I see that @khancyr suggested this change, but I don't see why. We should put includes in cpp files as much as we can.

OXINARF commented on libraries/AP_Common/AP_Time.h:
```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <AP_Common/AP_Common.h>
+#include <time.h>
```
I see that @khancyr suggested to put this here, but I don't see why. We should put includes in cpp files as much as we can.

rmackay9 replied:
Yes, I kind of thought so too.  I'll move it back.

OXINARF commented:
@rmackay9 Just made a comment, sorry for only seeing it now.

tridge commented:
I suggested AP_Common just on the principle of keeping AP_HAL as small as possible, but really it isn't that important. So If this passes travis then it can go in with either location.

rmackay9 commented:
I'm going to close this.  It's pretty old and I'm not sure we really care so much.
